### ISSUE TITLE
Compilation errors in Sentry Flutter due to Kotlin language version [Execution failed for task ':sentry_flutter:compileDebugKotlin'.]

### ISSUE THREAD

**Issue by giorgiogross** [getsentry/sentry-dart#2455]
### Platform

Flutter Mobile Android

### Obfuscation

Enabled

### Debug Info

Enabled

### Doctor
```
[✓] Flutter (Channel stable, 3.24.4, on macOS 15.1 24B83 darwin-arm64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.1)
[✓] VS Code (version 1.93.1)
[✓] Connected device (4 available)
[✓] Network resources

```

Might be related to #2396 

Since I upgraded my Kotlin (now 2.1.0) and Gradle (now 8.6.1) versions, sentry_flutter seems to have issues when compiling for Android. I can confirm this is an issue with this package, as by modifying the  `sentry_flutter/android/build.gradle` with
```
kotlinOptions {
        jvmTarget = JavaVersion.VERSION_1_8
 -      languageVersion = "1.4"
 +      languageVersion = "1.6"
}
```
I can compile my app again

### Version

8.10.1

### Steps to Reproduce

Upgrade to latest Kotlin + Gradle versions, then build app
-> compilation fails

### Expected Result

The app should compile and launch normally on Android

### Actual Result

Compilation fails with:
```
e: Language version 1.4 is no longer supported; please, use version 1.6 or greater.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':sentry_flutter:compileDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 15s
Error: Gradle task assembleDebug failed with exit code 1
```

### Are you willing to submit a PR?

None

**Comment by buenaflor** [comment]
hey thanks for the issue, we're looking into it. 

current issue for us is we don't know how many people a bump to `1.6` would affect - at least in a non-major version

**Comment by buenaflor** [comment]
Kotlin 1.5.31 was introduced as min version in Flutter `2.10.0` which is below what our minimum version is which is good.

In Flutter 3.0.0 (our min) a simple create app would generate a build with kotlin 1.6.10 which doesn't say much about the min version but I don't think users would deviate from that and downgrade - to me it seems safe to bump it now without a major.

@vaind wdyt?

**Comment by vaind** [comment]
agreed

**Comment by kuhnroyal** [comment]
I am getting the same error after bumping Kotlin to 2.1.0.

**Comment by buenaflor** [comment]
cc @kahest just fyi that we will bump this to `1.6.0` here

**Comment by fatmagani** [comment]
I am getting the same error (e: Language version 1.4 is no longer supported; please, use version 1.6 or greater.) after bumping Kotlin to 2.1.0 too

**Comment by vaind** [comment]
Should be fixed by https://github.com/getsentry/sentry-dart/pull/2456 and come in the next release

**Comment by filippo-orru** [comment]
TLDR: upgrade to `sentry_flutter: ^8.11.0` in your `pubspec.yaml` and run `pub get` ✅
